File: lists/Node.java
package lists;

public class Node<T> {
	public Node next;
	public Node prior;
	public T value;

    public Node(T value) {
        this.value = value;
    }

    public void setNext(Node next) {
        this.next = next;
    }
    public void setPrior(Node prior) {
	this.prior = prior;
    }
    public void setValue(T value) {
	this.value = value;
    }
    public T getValue() {
	return this.value;
    }
    public Node getNext() {
        return this.next;
    }
    public Node getPrior() {
        return this.prior;
    }
}

File: lists/python/Node.py
class Node:
	def __init__(self, value, next=None):
		self.value = value
		self.next = next

	def get_value(self):
		return self.value

	def set_value(self, value):
		self.value = value

	def get_next(self):
		return self.next

	def set_next(self, next):
		self.next = next


# Tests

def test_get_set_node():
	second_node = Node("B")
	head = Node("A", second_node)

	assert head.get_next() == second_node
	assert head.value == "A"

	head.set_value("Z")
	new_node = Node("C", second_node)
	head.set_next(new_node)

	assert head.value == "Z"
	assert head.get_next() == new_node

	next = head.get_next()
	assert next == new_node
	assert next.get_next() == second_node

	next = next.get_next()
	assert next == second_node
	assert next.get_next() is None

if __name__ == "__main__":
	test_get_set_node() 
